app/src/actions/tasks.ts

```diff
@@ -31,15 +33,24 @@ export function loadTasks() {
 export function addTask(task: Task, onComplete: Function) {
   return (dispatch) => {
     delete task._id;
-    return Tasks.add(task)
-      .subscribe((res) => {
-        if (res.length === 1) {
-          dispatch({
-            type: TASK_ADDED,
-            payload: res[0]
-          });
-        }
-      }, null, onComplete); }
+
+    const taskAdd$ = Tasks.add(task).publish();
+    taskAdd$.connect();
```

Comment:
Not sure this is necessary. The service returns an observable. You can subscribe to it right away or attach operators like Abdella suggested and then subscribe.